src/qr.jl

```diff
@@ -535,6 +535,19 @@ function ldiv!(A::QRCompactWY{T}, B::AbstractMatrix{T}) where {T}
     return B
 end
 
+"""
+    rank(A::QRPivoted{<:Any, T}; atol::Real=0, rtol::Real=min(n,m)*ϵ) where {T}
+
+Compute the numerical rank of the QR factorization `A` by counting how many diagonal entries of
+`A.factors` are greater than `max(atol, rtol*Δ₁)` where `Δ₁` is the largest calculated such entry.
+This is equivalent to the default [`rank(::AbstractMatrix)`](@ref) method except that it re-uses an existing QR factorization.
```

Comment:
It's not equivalent to the default `rank`, which uses an SVD, and may return a different value in marginal cases.

Reply:
Note also that computing the rank from QR factorization can be somewhat less reliable than doing so from SVD as noted in #1052 and in the linked discussion.  For example, there are pathological cases where the QR case can overestimate the rank as explained [here](https://discourse.julialang.org/t/how-to-find-the-linearly-independent-columns-rows-of-a-matrix/109772/24), though in most cases it seems to do okay.